crawl-ref/source/webserver/ws_handler.py

```diff
@@ -616,11 +643,17 @@ def go_lobby(self):
 
     def get_rc(self, game_id):
         if game_id not in config.games: return
-        with open(self.rcfile_path(game_id), 'r') as f:
-            contents = f.read()
+        path = self.rcfile_path(game_id)
+        try:
+            with open(path, 'r') as f:
+                contents = f.read()
+        # IOError for py2, OSError for py3
+        except (OSError, IOError):
+            contents = ''
         self.send_message("rcfile_contents", contents = contents)
 
     def set_rc(self, game_id, contents):
+        game_id = str(game_id) # protect against JS submitting bad data
```

Comment:
sorry, i don't understand this; could you describe in more detail? is it about preventing js sending non utf-8 data?

Reply:
I think the issue was that if you have a `game_id` of `0.23` it can get interpreted as a number in some parts of the client-side JS. But I'm not sure why I added this protection only to `set_rc` and not also to `get_rc` (or even setting it higher up, wherever this value is deserialised.

I'll remove it, if the problem re-appears I can add it as a new PR.

(I think the error case happens when creating RC files when you don't currently have an RC file, which only CPO runs into because the other DGL-enabled servers explicitly create blank RC files on user creation.)